api/src/main/java/org/apache/iceberg/RewriteFiles.java

```diff
@@ -34,13 +34,72 @@
  * will throw a {@link ValidationException}.
  */
 public interface RewriteFiles extends SnapshotUpdate<RewriteFiles> {
+  /**
+   * Delete a data file whose content was rewritten.
+   *
+   * @param dataFile a rewritten data file
+   * @return this for method chaining
+   */
+  default RewriteFiles deleteFile(DataFile dataFile) {
+    throw new UnsupportedOperationException(
+        this.getClass().getName() + " does not implement deleteFile");
+  }
+
+  /**
+   * Delete a delete file whose content was rewritten.
```

Comment:
Question: will this be not entirely accurate for RewritePositionDelete?  We may end up not rewriting invalid deletes.  Or is 'rewritten' more nuanced?

Reply:
Do you mean cases when we rewrite a position delete file but discard no longer valid deletes?